lib/compile.js

```diff
@@ -51,7 +51,7 @@ module.exports = {
 
     const configs = ensureArray(this.webpackConfig);
     const logStats = getStatsLogger(configs[0].stats, this.serverless.cli.consoleLog);
-    const concurrency = this.concurrency === undefined ? Infinity : this.concurrency;
+    const concurrency = this.concurrency === undefined ? Infinity : Number(this.concurrency);
```

Comment:
`Number(...)` or `parseInt(...)` may results in `NaN`, the following line should do better in error fallback.

```suggestion
    const concurrency = parseInt(this.concurrency) || Infinity;
```

Reply:
I think you're right, we should treat it as an error. I will move this check to Configuration ctor, which seems the best place to validate the config values (including your fallback to Infinity, which should probably be in DefaultConfig)

Reply:
Sounds good, I'll merge when you push to this branch.

Reply:
I found some time to do the update, see changes after I force-pushed.